bpo-37878: Make PyThreadState_DeleteCurrent() Internal (#15315)
python/cpython: Doc/c-api/init.rst

   The thread state must have been reset with a previous call to
   :c:func:`PyThreadState_Clear`.

.. c:function:: void PyThreadState_DeleteCurrent()

>     Destroy the current thread state and releases the global interpreter. The global

Reviewer: ```suggestion
    Destroy the current thread state and release the global interpreter lock.
```

>     interpreter lock need not be held. Like :c:func:`PyThreadState_Delete`, the global

Reviewer: ```suggestion
    Like :c:func:`PyThreadState_Delete`, the global
```

>     interpreter lock need not be held. The thread state must have been reset with a previous

Reviewer: ```suggestion
    interpreter lock need not be held. When using :c:func:`PyThreadState_DeleteCurrent`, the thread state must have been reset with a previous
```
